### ISSUE TITLE
Code coverage requires DebugType full

### ISSUE THREAD

**Issue by codito** [Microsoft/vstest#800]
For projects using the roslyn based project system, coverage requires `DebugType = Full` as shown below:

```xml
  <PropertyGroup>
    <TargetFramework>netcoreapp1.1</TargetFramework> 
    <DebugType>Full</DebugType>
  </PropertyGroup>
```

This issue tracks the requirement to support portable symbols in coverage infrastructure.

**Comment by smadala** [comment]
Making DebugType to Full for netcore breaks source navigation and context "Run Tests" option. related bug:https://github.com/Microsoft/vstest/issues/373

**Comment by cvs79** [comment]
When is this expected to be fixed out of the box. We cant advise clients to manually add <DebugType>Full</DebugType> to their projects to get this working. Also adding that setting shows a very strange code coverage list.  See screenshots

<img width="843" alt="code coverage" src="https://user-images.githubusercontent.com/8415874/32727001-4990a328-c87b-11e7-80d4-a32887f9d277.png">

**Comment by singhsarab** [comment]
There is a lot of feedback coming in around this.

**Comment by tmat** [comment]
Is this being worked on? What component needs to be updated to support Portable PDBs?

**Comment by smadala** [comment]
For timeline https://github.com/Microsoft/vstest/pull/1301#discussion_r151942331, Code coverage profiler need to support portable PDBs.

**Comment by Mohammed-ElSayed** [comment]
is there any update regarding this issue?
or is it still the fix that we need to in order to enable code coverage reports on TFS?

**Comment by max-buecker** [comment]
I am also genuinely interested in the status of this issue. For my netcore2.0 projects, I've added the debugtype full setting without any side-effects as far as I am aware. Is this still the way to go? Will there be a solution in the future, where this won't be required anymore?

**Comment by smadala** [comment]
@max-buecker Sorry for delay in fixing this issue.

CodeCoverage profiler has dependency on [msdia.dll](https://docs.microsoft.com/en-us/visualstudio/debugger/debug-interface-access/debug-interface-access-sdk) library, Currently msdia won't support for portable PDB. 
 
 Luckily @VCZYK Implemented to support portable with msdia. I have tested private bits on vstest repo, Code coverage works without any workaround. We are planning to ship it in VS 15.8.

**Comment by stephenegriffin** [comment]
The above workaround is correct for .net. For C++ and v141, full symbols is handled differently:
<GenerateDebugInformation>DebugFull</GenerateDebugInformation> 
See https://developercommunity.visualstudio.com/content/problem/17575/c-unit-testing-code-coverage-not-working.html for further details. This is a regression from v140, where the following is sufficient:
<GenerateDebugInformation>true</GenerateDebugInformation>

**Comment by HenrikFrystykNielsen** [comment]
FWIW, it seems that using pdbonly also does the trick:

    <DebugType>pdbonly</DebugType>

Hope this helps,

Henrik

**Comment by smadala** [comment]
Yesterday [VS 15.8 preview 3](https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2017-preview-relnotes#15.8_Preview3) released. It has fix for this issue.

Please try it and let us know if you find any issue(Please create separate issue with details repro steps), we will try to fix it ASAP.

**Comment by onovotny** [comment]
@smadala does that support pdb type of embedded?

**Comment by onovotny** [comment]
I just tried with 15.8 and it doesn't return any data for emedded pdb's, so this issue cannot be resolved yet.

**Comment by smadala** [comment]
@onovotny No, embedded pdb not supported. Currently plan is support it in VS 16.

**Comment by onovotny** [comment]
Is there a global MSBuild Property set when using VS's "Analyze Code Coverage command"? At the very least, it would let us set a conditional property on that so we can use portable for those builds.

**Comment by smadala** [comment]
>Is there a global MSBuild Property set when using VS's "Analyze Code Coverage command"?

@onovotny Sorry, No such property available.

**Comment by onovotny** [comment]
@smadala Can that be added for 15.8? It would seem to be useful to know if a build is happening in the context of prepping to run code coverage that way.

**Comment by smadala** [comment]
@onovotny AFAIK There is no infra/way to add the MSBuild property. Test explorer `Analyze Code Coverage command` comes into the picture only after build completed. 

@ManishJayaswal @AbhitejJohn  better persons to answer it?

**Comment by AbhitejJohn** [comment]
@onovotny : There is the [DesignMode switch](https://github.com/Microsoft/vstest-docs/blob/master/docs/configure.md) that the platform sets in the runsettings when running from IDE. Would that work?

**Comment by smadala** [comment]
Tracking embedded pdb https://github.com/Microsoft/vstest/issues/1748

**Comment by alastairtree** [comment]
From the looks of things this is not really closed, just bumped off to another issue, and still not fixed. Version 2.1 of .net core and would really expect `dotnet test --collect:"Code Coverage"` to just work out of the box. Please can we get some clear comms on how/when code coverage will work?

**Comment by smadala** [comment]
Install dotnet core sdk >= 2.1.400 from [here](https://www.microsoft.com/net/learn/get-started-with-dotnet-tutorial) and update `Microsoft.NET.Test.Sdk` to 15.8.0.

Example: https://github.com/pvlakshm/Samples/tree/master/NetCoreSampleForCC

**Comment by GaTechThomas** [comment]
This issue is referenced from a MS repo-based document:

https://github.com/Microsoft/vstest-docs/blob/master/docs/analyze.md

...which is referenced by official "dotnet test" documentation:

https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test#options

It would be helpful to have a final guidance on this thread (so that we don't have to dig further) and to lock the thread so that the final answer remains at the bottom.
